notebook/services/contents/filemanager.py

```diff
@@ -489,16 +489,8 @@ def delete_file(self, path):
         path = path.strip('/')
         os_path = self._get_os_path(path)
         rm = os.unlink
-        if os.path.isdir(os_path):
-            listing = os.listdir(os_path)
-            # Don't delete non-empty directories.
-            # A directory containing only leftover checkpoints is
-            # considered empty.
-            cp_dir = getattr(self.checkpoints, 'checkpoint_dir', None)
-            for entry in listing:
-                if entry != cp_dir:
-                    raise web.HTTPError(400, u'Directory %s not empty' % os_path)
-        elif not os.path.isfile(os_path):
+        
+        if not os.path.isfile(os_path):
```

Comment:
Previously this was checking (in pseudocode):

```python
if isdir():
    ...
elif isfile():
    error
```

Now it only checks `isfile()`, so directories hit the error case.

Reply:
sorry, the second check is `elif not isfile()`.

i.e. the error should occur if the path is neither a file nor a directory. With your changes, it occurs if it's not a file.

Reply:
Okay, thanks. I'll make the changes and send in a new PR.